sirepo/package_data/static/js/raydata.js

```diff
@@ -122,10 +143,40 @@ SIREPO.app.factory('raydataService', function(appState, requestSender, simulatio
     return self;
 });
 
-SIREPO.app.controller('AnalysisController', function($scope) {
+SIREPO.app.controller('AnalysisController', function(appState, persistentSimulation, raydataService, timeService, $scope) {
     const self = this;
     self.modelKey = 'analysisStatus';
     self.simScope = $scope;
+
+    self.simComputeModel = 'pollBlueskyForScansAnimation';
+
+    self.simState = persistentSimulation.initSimulationState(self);
+
+    self.simHandleStatus = function(data) {
+	// When not running we don't want to update the scans.
+	// There may be scans that the user has removed but are
+	// still being sent from the backend (in parallelStatus).
+	if (data.state !== 'running' || ! data.scans) {
+	    return;
+	}
+	const u = [];
+	data.scans.forEach((s) => {
+	    if (appState.models.scans.selected[s.uid]) {
+		return;
+	    }
+	    raydataService.selectOrDeselectScan(s);
```

Comment:
I found selectOrDeselect to be a bit confusing, and here, it is used just to select, and the abstraction has to be broken to know that (line 164 above). Seems to me that you might want the toggle code to be simpler for the checkbox, and to have a select and a deleselect, e.g.

```js
if (s.selected) {
    select(s)
}
else {
    deselect(s)
}
```

And then the select and delesect functions would be simpler, too. And the use here would be same.

Not sure. Just noting.

Reply:
I agree that it is confusing. There is a fair bit of state (and caching) that needs to be organized. I think spreading that across two functions may lead to bugs. It is clearer to me with an if/else that you need to do one thing in the select case and reverse it in the deselect case. If it is across functions one of the functions may be outside of my view and I would forget to update it. 

I think your code also breaks the abstraction some knowing that there is a "selected" field on the scan (although the checkbox code needs to know this so the cat is already out of the bag). I tried 0d4a6a78a which may be just more confusing. Not sure.